### ISSUE TITLE
Listener onPointerDown(_) in Stack with translucent-hitBehavior stopped working correctly in recent update.

### ISSUE THREAD

**Issue by SahajRana** [flutter/flutter#74606]
Hi Flutter team,

`Listener` and `RawGestureDetector` in `Stack` with translucent hitBehavior stopped working correctly in the recent update.

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, 1.26.0-13.0.pre.140, on Microsoft Windows [Version 10.0.18362.1082], locale en-IN)
[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.6.3)
[√] Android Studio (version 3.1)
[√] Android Studio (version 4.0)
[√] VS Code, 64-bit edition (version 1.48.0)
[√] Connected device (3 available)

! Doctor found issues in 1 category.
```

Flutter code:

```
return Stack(
          children: [
            Listener(
              behavior: HitTestBehavior.translucent,
              onPointerDown: (_){
                print("onTAP 2");
              },
              child: Container(
                width: 100,
                height: 100,
                color: Color(0x1F000000),
              ),
            ),
           Listener(
              behavior: HitTestBehavior.translucent,
              onPointerDown: (_){
                print("onTAP 1");
              },
              child: Container(
                width: 100,
                height: 100,
                color: Color(0x1F000000),
              ),
            ),
            Listener(
              behavior: HitTestBehavior.translucent,
              onPointerDown: (_){
                print("onTAP 0");
              },
              child: Container(
                width: 100,
                height: 100,
                color: Color(0x1F000000),
              ),
            ),
          ]
      );
```

I only receive 
```
flutter: onTAP 0
```

**Comment by iapicca** [comment]
@SahajRana 
I can confirm this behavior, as for my [stackoverflow question](https://stackoverflow.com/questions/64856179/how-do-i-allow-gestures-to-pass-through-a-widget-in-a-stack),
but I don't recall this behavior being different in previous versions,
in what `flutter version` did you experience a different behavior?

**Comment by SahajRana** [comment]
@iapicca, I made a mistake here, I meant with the `Listener` widget, not the GestureDetector. `onPointerDown(_)` used to work in this case for `Flutter-v1.21.0_8.0.pre.140`.

**Comment by SahajRana** [comment]
@justinmc, @dnfield, @jonahwilliams, @Piinks, @xu-baolin, It would be great if you guys could help me with this.

**Comment by pedromassangocode** [comment]
Hi @SahajRana 
Please provide your `flutter doctor -v`.
Thank you

**Comment by SahajRana** [comment]
I have attached it at the start of the question as well.
```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, 1.26.0-13.0.pre.140, on Microsoft Windows [Version 10.0.18362.1082], locale en-IN)
[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.6.3)
[√] Android Studio (version 3.1)
[√] Android Studio (version 4.0)
[√] VS Code, 64-bit edition (version 1.48.0)
[√] Connected device (3 available)

! Doctor found issues in 1 category.
```

**Comment by iapicca** [comment]
@SahajRana  @pedromassangocode 

testing  with [fvm](https://pub.dev/packages/fvm):

1. [install Dart SDK](https://dart.dev/get-dart#install)
2. [update your path](https://dart.dev/tools/pub/cmd/pub-global#running-a-script-from-your-path)
3. run 
```console
pub global activate fvm
flutter create issue_74606
cd issue_74606
fvm use 1.21.0-9.1.pre
```
<details>
<summary>logs</summary>

```bash
francesco@francesco-yoga720:~/projects/issue_74606$ fvm use 1.21.0-9.1.pre
Flutter 1.21.0-9.1.pre is not installed.
Would you like to install it? Y/n: y
Installing version: 1.21.0-9.1.pre
Cloning into '/home/francesco/fvm/versions/1.21.0-9.1.pre'...
remote: Enumerating objects: 1, done.        
remote: Counting objects: 100% (1/1), done.        
remote: Total 253205 (delta 0), reused 0 (delta 0), pack-reused 253204        
Receiving objects: 100% (253205/253205), 107.52 MiB | 357.00 KiB/s, done.
Resolving deltas: 100% (194867/194867), done.
Version installed: 1.21.0-9.1.pre
Project now uses Flutter: 1.21.0-9.1.pre

```

</details>

4. replace the content of `main.dart` with the sample below 
<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(
      MaterialApp(
        home: Material(
          child: Stack(
            children: [for (var i = 0; i < 3; i++) MyListener(i)],
          ),
        ),
      ),
    );

class MyListener extends StatelessWidget {
  MyListener(this.index) : super(key: ValueKey('MyListener: $index'));
  final int index;
  @override
  Widget build(BuildContext context) => Center(
        child: Listener(
          behavior: HitTestBehavior.translucent,
          onPointerDown: (_) => print('$index'),
          child: SizedBox.expand(),
        ),
      );
}

```

</details>

5. run `flutter run -v`
<details>
<summary>logs</summary>

```bash
[ +181 ms] Syncing files to device Pixel 3a...
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +2 ms] <- recompile package:issue_74606/main.dart 07930dad-781d-427a-9874-726914dcacd3
[        ] <- 07930dad-781d-427a-9874-726914dcacd3
[ +170 ms] Updating files.
[        ] DevFS: Sync finished
[   +2 ms] Syncing files to device Pixel 3a... (completed in 176ms)
[   +1 ms] Synced 0.0MB.
[   +1 ms] <- accept
[  +56 ms] Connected to _flutterView/0x76339fdce0.
[   +1 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[        ] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] An Observatory debugger and profiler on Pixel 3a is available at: http://127.0.0.1:44331/ZyQ6tOTXEY4=/
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[+245578 ms] I/flutter (31548): 2
[        ] I/flutter (31548): 1
[        ] I/flutter (31548): 0
[ +877 ms] I/flutter (31548): 2
[        ] I/flutter (31548): 1
[        ] I/flutter (31548): 0
[ +595 ms] I/flutter (31548): 2
[        ] I/flutter (31548): 1
[   +2 ms] I/flutter (31548): 0
[ +459 ms] I/flutter (31548): 2
[        ] I/flutter (31548): 1
[        ] I/flutter (31548): 0
[ +452 ms] I/flutter (31548): 2
[        ] I/flutter (31548): 1
[        ] I/flutter (31548): 0
[+7375 ms] Application finished.
[   +4 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +7 ms] Exit code 0 from: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[        ] 965AY0WP5C tcp:41361 tcp:37271
[   +1 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --remove tcp:41361

```

</details>

## [listener](https://api.flutter.dev/flutter/widgets/Listener-class.html) behavior is "**_pass through_**"

6. update project to the latest beta `1.25.0-8.3.pre`

```console
fvm use 1.25.0-8.3.pre
```

<summary>logs</summary>

```bash
francesco@francesco-yoga720:~/projects/issue_74606$ fvm use 1.25.0-8.3.pre
Flutter 1.25.0-8.3.pre is not installed.
Would you like to install it? Y/n: y
Installing version: 1.25.0-8.3.pre
Cloning into '/home/francesco/fvm/versions/1.25.0-8.3.pre'...
remote: Enumerating objects: 3, done.        
remote: Counting objects: 100% (3/3), done.        
remote: Compressing objects: 100% (3/3), done.        
remote: Total 279564 (delta 0), reused 0 (delta 0), pack-reused 279561        
Receiving objects: 100% (279564/279564), 123.10 MiB | 522.00 KiB/s, done.
Resolving deltas: 100% (216074/216074), done.
Version installed: 1.25.0-8.3.pre
Project now uses Flutter: 1.25.0-8.3.pre

```
</details>


7. run `flutter run -v`
<details>
<summary>logs</summary>

```bash
[ +110 ms] Syncing files to device Pixel 3a...
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +1 ms] <- recompile package:issue_74606/main.dart 234443e5-cdd8-4c13-bfad-c93401bc8279
[        ] <- 234443e5-cdd8-4c13-bfad-c93401bc8279
[  +96 ms] Updating files.
[        ] DevFS: Sync finished
[        ] Syncing files to device Pixel 3a... (completed in 100ms)
[        ] Synced 0.0MB.
[        ] <- accept
[  +65 ms] Connected to _flutterView/0x7633a005d0.
[   +7 ms] Flutter run key commands.
[   +6 ms] r Hot reload. 🔥🔥🔥
[   +4 ms] R Hot restart.
[   +1 ms] h Repeat this help message.
[   +1 ms] d Detach (terminate "flutter run" but leave application running).
[   +1 ms] c Clear the screen
[   +1 ms] q Quit (terminate the application on the device).
[        ] An Observatory debugger and profiler on Pixel 3a is available at: http://127.0.0.1:38717/Cy_6TYLqLuE=/
[   +1 ms] 
           Flutter DevTools, a Flutter debugger and profiler, on Pixel 3a is available at: http://127.0.0.1:9100?uri=http%3A%2F%2F127.0.0.1%3A38717%2FCy_6TYLqLuE%3D%2F
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[+16301 ms] I/flutter ( 6460): 2
[        ] I/flutter ( 6460): 1
[   +3 ms] I/flutter ( 6460): 0
[ +802 ms] I/flutter ( 6460): 2
[        ] I/flutter ( 6460): 1
[        ] I/flutter ( 6460): 0
[ +448 ms] I/flutter ( 6460): 2
[        ] I/flutter ( 6460): 1
[   +3 ms] I/flutter ( 6460): 0
[ +306 ms] I/flutter ( 6460): 2
[        ] I/flutter ( 6460): 1
[        ] I/flutter ( 6460): 0
[ +272 ms] I/flutter ( 6460): 2
[        ] I/flutter ( 6460): 1
[        ] I/flutter ( 6460): 0
[ +319 ms] I/flutter ( 6460): 2
[        ] I/flutter ( 6460): 1
[        ] I/flutter ( 6460): 0
[ +346 ms] I/flutter ( 6460): 2
[        ] I/flutter ( 6460): 1
[        ] I/flutter ( 6460): 0
[+3880 ms] I/flutter ( 6460): 2
[        ] I/flutter ( 6460): 1
[   +2 ms] I/flutter ( 6460): 0
[ +368 ms] I/flutter ( 6460): 2
[        ] I/flutter ( 6460): 1
[        ] I/flutter ( 6460): 0
[+12908 ms] DevFS: Deleting filesystem on the device (file:///data/user/0/com.example.issue_74606/code_cache/issue_74606YHEDFN/issue_74606/)
[  +39 ms] DevFS: Deleted filesystem on the device (file:///data/user/0/com.example.issue_74606/code_cache/issue_74606YHEDFN/issue_74606/)
[ +413 ms] Service protocol connection closed.
[        ] Application finished.
[   +3 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +6 ms] Exit code 0 from: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[        ] 965AY0WP5C tcp:41361 tcp:37271
           965AY0WP5C tcp:43689 tcp:43217
[   +1 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --remove tcp:41361
[   +8 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --remove tcp:43689
[   +6 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +7 ms] Exit code 0 from: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +1 ms] "flutter run" took 98,679ms.
[ +255 ms] ensureAnalyticsSent: 252ms
[   +3 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[  +11 ms] Shutdown hooks complete
[   +1 ms] exiting with code 0

```

</details>

the latest beta still shows the same behavior as the previous

**Comment by SahajRana** [comment]
@iapicca, thanks for running the code. Have you tried it for `Channel master, 1.26.0-13.0.pre.140`. It seems to have stopped working from this version.

**Comment by SahajRana** [comment]
@iapicca, Well it's weird or maybe intended. I tried it with SizedBox like your example, it seems to work fine. 
But **If you try it with `Container` widget with `color` property given, it stops "pass-through".**

**Comment by iapicca** [comment]
@SahajRana 
I can confirm that the modified code sample behave as you described
<details>
<summary>modified code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(
      MaterialApp(
        home: Material(
          child: Stack(
            children: [for (var i = 0; i < 3; i++) MyListener(i)],
          ),
        ),
      ),
    );

class MyListener extends StatelessWidget {
  MyListener(this.index) : super(key: ValueKey('MyListener: $index'));
  final int index;
  @override
  Widget build(BuildContext context) => Center(
        child: Listener(
          behavior: HitTestBehavior.translucent,
          onPointerDown: (_) => print('$index'),
          child: Container(
            color: const Color.fromRGBO(200, 0, 0, .3),
          ),
        ),
      );
}
```

</details>

nevertheless this behavior doesn't seem to change with previous versions

##

testing using [fvm](https://pub.dev/packages/fvm) with the following steps:

1. [install Dart SDK](https://dart.dev/get-dart#install)
2. [update your path](https://dart.dev/tools/pub/cmd/pub-global#running-a-script-from-your-path)
3. run
```console
pub global activate fvm
flutter create issue
cd issue
fvm use 1.20.4
```

 `1.20.4` is the closest `stable release` precedent to `1.21.0_8.0.pre.140` referenced [here](https://github.com/flutter/flutter/issues/74606#issuecomment-766449422)

<details>
<summary>flutter version</summary>

```bash
Flutter 1.20.4 is not installed.
Would you like to install it? Y/n: y
Installing version: 1.20.4
Cloning into '/home/francesco/fvm/versions/1.20.4'...
remote: Enumerating objects: 246506, done.        
remote: Total 246506 (delta 0), reused 0 (delta 0), pack-reused 246506        
Receiving objects: 100% (246506/246506), 104.35 MiB | 1.71 MiB/s, done.
Resolving deltas: 100% (189550/189550), done.
Version installed: 1.20.4
Project now uses Flutter: 1.20.4
```

</details>


<details>
<summary>logs</summary>

```bash
[ +156 ms] Syncing files to device Pixel 3a...
[   +2 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +2 ms] <- recompile package:issue/main.dart 275d8ecc-db5e-4c09-8337-4132d1f9c743
[        ] <- 275d8ecc-db5e-4c09-8337-4132d1f9c743
[ +116 ms] Updating files.
[        ] DevFS: Sync finished
[        ] Syncing files to device Pixel 3a... (completed in 121ms)
[        ] Synced 0.0MB.
[        ] <- accept
[  +86 ms] Connected to _flutterView/0x7633a09100.
[   +3 ms] Flutter run key commands.
[   +4 ms] r Hot reload. 🔥🔥🔥
[   +3 ms] R Hot restart.
[   +1 ms] h Repeat this help message.
[   +1 ms] d Detach (terminate "flutter run" but leave application running).
[   +1 ms] c Clear the screen
[   +1 ms] q Quit (terminate the application on the device).
[   +1 ms] An Observatory debugger and profiler on Pixel 3a is available at: http://127.0.0.1:38115/KTvh1FheGXc=/
[   +3 ms] 
           Flutter DevTools, a Flutter debugger and profiler, on Pixel 3a is available at: http://127.0.0.1:9100?uri=http%3A%2F%2F127.0.0.1%3A38115%2FKTvh1FheGXc%3D%2F
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[+2364 ms] I/flutter (14723): 2
[+1814 ms] I/chatty  (14723): uid=10410(com.example.issue) 1.ui identical 1 line
[   +1 ms] I/flutter (14723): 2
[+112560 ms] DevFS: Deleting filesystem on the device (file:///data/user/0/com.example.issue/code_cache/issueOIJLWG/issue/)
[  +27 ms] DevFS: Deleted filesystem on the device (file:///data/user/0/com.example.issue/code_cache/issueOIJLWG/issue/)
[ +276 ms] Service protocol connection closed.
[        ] Application finished.
[   +3 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +5 ms] Exit code 0 from: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[        ] 965AY0WP5C tcp:42943 tcp:40147
[   +1 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --remove tcp:42943
[   +9 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +8 ms] Exit code 0 from: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +1 ms] "flutter run" took 158,293ms.
[ +258 ms] ensureAnalyticsSent: 253ms
[   +5 ms] Running shutdown hooks
[   +1 ms] Shutdown hook priority 4
[   +3 ms] Shutdown hooks complete
[   +1 ms] exiting with code 0

```

</details>

**Comment by SahajRana** [comment]
@iapicca, well everything is working fine now. This confusion aroused to me because of the `MouseRegion` widget.
With the latest flutter version the `onPointerHover(_)` and `onPointerExit(_)` function in `Listener` widget is moved to `MouseRegion` widget respective as `onHover(_)` and `onExit(_)`. 

But `MouseRegion` widget doesn't have `HitTestBehavior`, but just `opaque: bool` property which is set to `true` by default so my previous "pass-through" code stopped working. So the fix was to set `MouseRegion` widget  `opaque: bool` property to `false`.

PS: I'm building a windows app, so it needs onHover.
